src/cpu/flexcpu/inflight_inst.hh

```diff
@@ -644,6 +647,14 @@ class InflightInst : public ExecContext,
 
     void pipeTrace();
 
+    // Since there are predicated instructions, the producer of a register
+    // might be predicated to false, and we should not forward anything
+    // from the instruction. This function determines the instruction holding
+    // the valid result from the register dependency chain. If no such an
+    // instruction exists, the backing context is holding the valid result.
+    std::pair<std::shared_ptr<InflightInst>, int>
+    getRegProducer(const StaticInst* si, int op_idx) const;
```

Comment:
Let's talk about this too... I think we may need to accelerate this with a change in the dependency map.